rusk-wallet/src/bin/io/prompt.rs

```diff
@@ -316,149 +296,101 @@ pub(crate) fn request_gas_price(
         min_gas_price
     };
 
-    let default_gas_price = Dusk::from(default_gas_price).into();
-    let min_gas_price = Dusk::from(min_gas_price).into();
-
-    let question = requestty::Question::float("amt")
-        .message("Introduce the gas price for this transaction:")
-        .default(default_gas_price)
-        .validate_on_key(|f, _| {
-            check_valid_denom(f, MIN_CONVERTIBLE, MAX_CONVERTIBLE).is_ok()
-        })
-        .validate(|f, _| check_valid_denom(f, MIN_CONVERTIBLE, MAX_CONVERTIBLE))
-        .validate(|f, _| {
-            if f < min_gas_price {
-                Err("Gas limit too low".to_owned())
-            } else {
-                Ok(())
-            }
-        })
-        .build();
-
-    let a = requestty::prompt_one(question)?;
-    let price = Dusk::from(a.as_float().expect("answer to be a float"));
-    Ok(*price)
+    request_token(
+        "gas price",
+        Dusk::from(min_gas_price),
+        MAX_CONVERTIBLE,
+        Some(default_gas_price as f64),
+    )
+    .map(|dusk| *dusk)
 }
 
 pub(crate) fn request_str(
     name: &str,
     max_length: usize,
 ) -> anyhow::Result<String> {
-    let question = requestty::Question::input("string")
-        .message(format!("Introduce string for {}:", name))
-        .validate(|input, _| {
-            if input.len() > max_length {
-                Err(format!(
-                    "Input exceeds the maximum length of {} characters",
-                    max_length
-                ))
-            } else {
-                Ok(())
-            }
-        })
-        .build();
-
-    let a = requestty::prompt_one(question)?;
-    Ok(a.as_string().expect("answer to be a string").to_owned())
+    Ok(
+        Text::new(format!("Introduce string for {}:", name).as_str())
+            .with_validator(move |input: &str| {
+                if input.len() > max_length {
+                    Ok(Validation::Invalid(
+                        format!(
+                            "Input exceeds the maximum length of {} characters",
```

Comment:
I'm not seeing this validation error pop up, see:
![image](https://github.com/user-attachments/assets/ee160dad-2fad-4d09-9dcb-8762f0efbed3)

Before, it would just stop you from adding any more characters.

This sentence also doesn't have an ending `.`, while some other validation errors do.